app/serializers/serialize_submission.rb

```diff
@@ -4,12 +4,20 @@ class SerializeSubmission
   initialize_with :submission
 
   def call
+    return unless submission
+
     {
       uuid: submission.uuid,
       tests_status: submission.tests_status,
       links: {
-        cancel: Exercism::Routes.api_submission_cancellations_url(submission, auth_token: user.auth_tokens.first.to_s)
-      }
+        cancel: Exercism::Routes.api_submission_cancellations_url(submission, auth_token: user.auth_tokens.first.to_s),
+        submit: Exercism::Routes.api_solution_iterations_url(
+          submission.solution.uuid,
+          submission_id: submission.uuid,
+          auth_token: user.auth_tokens.first.to_s
+        )
+      },
+      test_run: SerializeSubmissionTestRun.(submission.test_runs.last)
```

Comment:
We also add the last test run serialized here.

Reply:
Having considered this more, I'm pretty sure I don't think we should do this. Because this will get added to all websockets hooks on this, and test run results can be very large (like 1MB if something bad happens).

So I think we have an extra endpoint instead that React looks up to get the last test run. `/api/submissions/XYZ/test_run` would work. And that should output 
```{test_run: SerializeSubmissionTestRun.(submission.test_runs.last)```

Reply:
I agree with making it a regular call. It's perfectly fine (and a good idea) to have websockets "ping", called notify to refresh/invalidate -- which then makes this call a regular REST api.

Reply:
Thanks! Will do that :)